app/src/ui/app-error.tsx

```diff
@@ -198,9 +286,17 @@ export class AppError extends React.Component<IAppErrorProps, IAppErrorState> {
   }
 
   private renderFooter(error: Error) {
-    const e = error instanceof ErrorWithMetadata ? error.underlyingError : error
+    if (isErrorWithMetaData(error)) {
+      const metaDataFooter = this.renderErrorWithMetaDataFooter(error)
 
-    if (e instanceof GitError) {
+      if (metaDataFooter) {
+        return metaDataFooter
+      }
+    }
+
+    const e = getUnderlyingError(error)
+
+    if (isGitError(e)) {
```

Comment:
It took me a while to understand this logic, it seems like L297-L303 is doing the same as L105-L109 when the error is `ErrorWithMetadata`, but in that case if I understand the logic correctly `metaDataFooter` will not be `null` (it can never be `null`) and this code won't get executed.

If I understood the whole thing correctly, I think we can simplify this whole thing by removing the `renderErrorWithMetaDataFooter()` method and changing this method to something like:

```jsx
  private renderFooter(error: Error) {
    if (isCloneError(error)) {
      return this.renderRetryCloneFooter() // The renderRetryCloneFooter() doesn't need any param
    }

    const e = getUnderlyingError(error)

    if (isGitError(e)) {
      return this.renderGitErrorFooter(e)
    }

    return this.renderDefaultFooter()
  }
```

Reply:
Yeah, this was another misguided attempt at a generalization of the AppError component to handle even more errors in the future. I've backed away from that now  given that thus far there's only two special kinds of errors and when we add more we'll revisit proper refactoring